third-party/crowdsec/README.md

```diff
@@ -0,0 +1,67 @@
+# James' extensions for Crowdsec
+
+This module is for developing and delivering extensions to James for the [Crowdsec](https://www.crowdsec.net/) (IP filtering system)
+
+## How to run
+
+- The Crowdsec extension requires an extra configuration file `crowdsec.properties` to configure Crowdsec connection
+  Configuration parameters:
+    - `crowdsecUrl` : URL defining the Crowdsec's bouncer. Eg: http://crowdsec:8080/v1
+    - `apiKey` : Api key for pass authentication when request to Crowdsec's bouncer.
+
+- Declare the `extensions.properties` for this module.
+
+```
+guice.extension.module=org.apache.james.module.CrowdsecModule
+```
+
+- Declare the Crowdsec EhloHook in `smtpserver.xml`. Eg:
+
+```
+<handlerchain>
+    <handler class="org.apache.james.smtpserver.fastfail.ValidRcptHandler"/>
+    <handler class="org.apache.james.smtpserver.CoreCmdHandlerLoader"/>
+    <handler class="org.apache.james.CrowdsecEhloHook"/>
+</handlerchain>
+```
+
+- Docker compose file example: [docker-compose.yml](docker-compose.yml).
+- The sample-configuration: [sample-configuration](sample-configuration)
+- For running docker-compose, first compile this project
+
+```
+mvn clean install -DskipTests
+```
+then run it: `docker-compose up`
+
+## Additional webadmin endpoints
+
+### Check IP blocked by Crowdsec (decisions)
+
+Crowdsec will expose port 8080 for queries to get the list of IP addresses blocked (get decisions)
+
+```bash
+curl -XGET http://localhost:8082/v1/decisions -H "X-Api-Key: default_api_key" -H 'accept: application/json' | jq .
```

Comment:
8080 then below 8082 (I know the exposed port changed, but still misleading).

BTW CrowdSec just needs to stay behind James, no need to expose CrowdSec service to host machine, no?